### ISSUE TITLE
Question: Manipulating original arrays

### ISSUE THREAD

**Issue by jusack** [jonniedie/ComponentArrays.jl#72]
In the main readme there is this phrase that says:

> This will create a new ComponentArray whose data is a view into the original, ...

I thought this sentence meant that I can operate on the ComponentArray and will also manipulate the original arrays. However this seems not to be the case:
```julia
v1 = [1,1,1]
v2 = [2,2,2]

c = ComponentArray(v1=v1, v2=v2)

c[1] = 0 # change ComponentArray
c.v1 == v1 #false, change did not happen to the original array
```

Is there a way to achieve this behaviour or is it just not the inteded purpose of this type?

**Comment by jonniedie** [comment]
Ohhh, yeah I should be more clear on that. This is saying that sub-`ComponentArray`s will have views back into the parent `ComponentArray`. There isn't a way to keep views back into the data that the `ComponentArray` was first made from because everything is backed by a single array. When I first started on this package, I experimented with having things backed by separate arrays, but landed on the current design for a number of reasons.

`ArrayPartition` from RecursiveArrayTools.jl is a little better for this purpose. The only thing is you'd be giving up named array partions like you'd have with a `ComponentArray`.

```julia
julia> v1 = [1,1,1];

julia> v2 = [2,2,2];

julia> c = ArrayPartition(v1, v2)
([1, 1, 1], [2, 2, 2])

julia> c[1] = 0
0

julia> c.x[1] == v1
true
```
